src/server/game/Movement/MotionMaster.cpp

```diff
@@ -382,6 +382,8 @@ void MotionMaster::MoveJump(float x, float y, float z, float speedXY, float spee
     init.MoveTo(x, y, z, false);
     init.SetParabolic(max_height, 0);
     init.SetVelocity(speedXY);
+    if (HasDst)
```

Comment:
nans are evil, would rather see optional used here BUT given that orientation logically belongs after Z (wouldnt make sense to make it last arg) i think your solution of using bool with default value as last argument is perfect.